Allow appending to existing arrays, force explicit null value allow for mapping from keys with null value.

ccelson opened the pull request:
I'm not sure if this is too specific to our use case, but we found that it was tricky when we wanted to push items onto a newly mapped array, we would constantly be writing over the initial transform. This change allows you to specify 

```javascript
source[]:{
  key:'destination[]+',
  transform: (val)=> mappingFunction(val)
},
someOtherSource[]:{
  key:'destination[]+',
  transform: (val)=> mappingFunction(val)
}

outputs

{
  destination:[
    {/*mappedObject from mappingFunction*/},
    {/*other mappedObject from the mappingFunction*/}
  ]
}
```
Which would then append the value to the existing array.

We also wanted to minimize the footprint of our transformation, so we wanted null values to be excluded in the mapping, unless explicitly specified with an postfix like opertator

```javascript
var sourceObject = {
  source:null,
  source2:null
},
mappingObject = {
  source:destinationMapNullValues?,
  source2:desinationDoNotMapNull
};

The result of the transform is
{
  destinationMapNullValues:null
}
```

wankdanker commented:
This is looking pretty good. I like the additions. In addition to the .exec()/.test() suggestion I made, could you please update the README to include the new features. Thanks!

ccelson commented:
Thanks for the feedback, glad you think it could work. If you have any suggestions to change the readme entries I added please let me know and I'll make the changes! I just changed the one null regex handler to regex.test, since the array append functionality uses the matches

wankdanker commented:
Just out of curiosity, did you have any reasoning for the '[]+' notation?

I was thinking that not knowing what may be added to this module in the future, it might be better to have things that apply to the array inside the brackets or '[+]'. That way theoretically '[+]+' could mean something (I don't know what). I just want to consider the future compatibility of introducing new features, because once we adopt it, it will probably stick around and be a pain to change once it's being used by people.

I bet '[]+' is way easier to implement though. What are your thoughts?

ccelson commented:
That could work fine, I believe we'd only need to change the regex right?
to this instead /(\[\]|\[\+\])$/g   ? I could do that if thats the route you want to go. I see your point

ccelson commented:
Derp, sorry just went to go look at this and realized the problem. Yea I'm if the added complexity of having a + gives the project that much more value? It's up to you, I think it could still be done with some more complex conditional logic.

To me personally I like the array[]+ version, only because it to me reads more naturally that I want to add something to the end of the array.

ccelson commented:
Any thoughts here @wankdanker ? Thanks!

wankdanker commented:
Hey @ccelson, Sorry, been swamped, this fell off my radar. array[]+ is fine. I'll get this merged asap.

wankdanker commented:
@ccelson, published object-mapper@3.0.0 to npm. I had to do a major version bump because some of the pre-existing tests were modified to pass (I checked out the tests from v2.2.5 and 3 failed).

Thank you for your contribution!
